azurerm/internal/services/cdn/parse/cdn_endpoint_custom_domain.go

```diff
@@ -0,0 +1,32 @@
+package parse
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type CdnEndpointCustomDomainId struct {
+	ResourceGroup string
+	ProfileName   string
+	EndpointName  string
+	Name          string
+}
+
+func CdnEndpointCustomDomainID(input string) (*CdnEndpointCustomDomainId, error) {
+	// Example ID: /subscriptions/<sub1>/resourcegroups/<grp1>/providers/Microsoft.Cdn/profiles/<profile1>/endpoints/<ep1>/customdomains/<domain1>
+	groups := regexp.MustCompile(`^(.+)/customdomains/([^/]+)$`).FindStringSubmatch(input)
```

Comment:
rather than using a regex, can we use `azure.ParseResourceID` here, that provides more granular errors?

Reply:
Hi @tombuildsstuff 
I intend to reuse the parse function for cdn endpoint here, otherwise we will have to duplicate the cdn endpoint part parsing code here.